src/main/java/org/testng/internal/ClonedMethod.java

```
@@ -147,6 +148,21 @@ public IRetryAnalyzer getRetryAnalyzer() {
     return m_method.getRetryAnalyzer();
```

you should throw an exception if the method is called because it is not supposed to be the case
Same for the setter

---

@juherr - That basically is as good as us removing the methods without deprecation. I believe the whole notion of deprecation is to just warn users that an API is going to be removed but still work. Throwing exceptions will cause the functionality to cease working and need code change by downstream users. If that's what we need then why use a deprecation strategy ? I am not sure I get this.